From 1e55618a24da234d8d3e629d9ec942e6026f7d94 Mon Sep 17 00:00:00 2001
From: Rahul Sandhu <nvraxn@gmail.com>
Date: Sun, 30 Nov 2025 00:30:10 +0000
Subject: [PATCH 16/17] gcc_config_t: allow reading cgroup files for cpu.max

Various coreutils, such as sort, now attempt[1] to read cpu.max due to
a change in gnulib[2]:

> avc:  denied  { read } for  pid=22163 comm="sort" name="cpu.max" dev="cgroup2" ino=1808 scontext=superuser:sysadm_r:gcc_config_t tcontext=system_u:object_r:cgroup_t tclass=file permissive=0

[1] https://gitweb.git.savannah.gnu.org/gitweb/?p=coreutils.git;a=commit;h=c4df55be7f78d29415abd7765f863ff38184f86e
[2] https://gitweb.git.savannah.gnu.org/gitweb/?p=gnulib.git;a=commit;h=9b07115f4a344effef1dde8bd0e6e356d4b0e744

Signed-off-by: Rahul Sandhu <nvraxn@gmail.com>
---
 policy/modules/admin/portage.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/admin/portage.te b/policy/modules/admin/portage.te
index a665d3d52..f8fe36e90 100644
--- a/policy/modules/admin/portage.te
+++ b/policy/modules/admin/portage.te
@@ -128,6 +128,10 @@ files_search_runtime(gcc_config_t)
 # the directory it is being run from
 files_list_all(gcc_config_t)
 
+# for sort/other coreutils:
+# https://gitweb.git.savannah.gnu.org/gitweb/?p=coreutils.git;a=commit;h=c4df55be7f78d29415abd7765f863ff38184f86e
+fs_read_cgroup_files(gcc_config_t)
+
 # seems to be ok without this
 init_dontaudit_read_script_status_files(gcc_config_t)
 
-- 
2.47.3

